tests/lib/Output/GeneratorTest.php

```diff
@@ -185,6 +185,27 @@ public function testNonEmptyDocument()
 
         $this->assertFalse($generator->isEmpty());
     }
+
+    abstract protected function assertSnapshot(string $snapshotName, string $generatedContent): void;
+
+    public function testStartValueElementWithAttributes(): void
+    {
+        $generator = $this->getGenerator();
+        $generator->startDocument('test');
+        $generator->startObjectElement('Element');
+        $generator->startValueElement(
+            'element',
+            'value',
+            [
+                'attribute1' => 'attribute_value1',
+                'attribute2' => 'attribute_value2',
+            ]
```

Comment:
Please, add different types of values. Floats, ints, booleans, nulls. Both as element value and attributes.

Reply:
> Please, add different types of values. Floats, ints, booleans, nulls. Both as element value and attributes.

@Steveb-p added via [e5cedf7](https://github.com/ibexa/rest/pull/65/commits/e5cedf795cb5a3e11af262d64b352ed405daca17).
The results for boolean are quite peculiar.

for `attribute1: true`, `attribute2: false`, `element value: false`, we get:
```xml
<element attribute1="1" attribute2=""/>
```
Instead of 
```xml
<element attribute1="true" attribute2="false">false</element>
```

This could be solved by `if (is_bool($value)) $value = $this->serializeBoolean($value)` but I feel like that would be big BC break.

Reply:
Figured, I had the exact same behavior in Ibexa Connect package, where I also had to conditionally apply the boolean serialization.

Considering it would be a BC break we cannot do it now of course, but it's something to keep in mind for 5.0.